### PR TITLE
chore: allow renovate @types/node update 

### PR DESCRIPTION
**Pull request by jerome-benoit**
It's a dev dep.

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: renovate.json [hunk 1]
```diff
@@ -34,6 +33,14 @@
...
       "rangeStrategy": "bump"
     },
+    {
+      "matchDepTypes": ["devDependencies"],
+      "groupName": "all major dev dependencies",
+      "groupSlug": "all-major-dev",
+      "matchUpdateTypes": [
+        "major"
+      ]
+    },
```
Comment (voxpelli) [inline renovate.json:43]: I prefer to not group major dev dependencies as the major bump indicate that they are breaking in some way and bundling many such together can make it harder to merge as a blocking breaking change from any one dependency will block the merging of them all

File: renovate.json [hunk 2]
```diff
@@ -24,7 +24,6 @@
   "supportPolicy": [
     "all"
   ],
-  "ignoreDeps": ["@types/node"],
```
Comment (voxpelli) [inline renovate.json:27]: Good call 👍 Its covered by `helpers:disableTypesNodeMajor` anyway :)

**Review by voxpelli — CHANGES_REQUESTED** [review]
> Dedupe package-lock.json: renovate lock file maintenance PR merge has
introduced typescript-eslint related deps duplicates

This is odd, as Renovate [lock file maintenance](https://docs.renovatebot.com/configuration-options/#lockfilemaintenance) is described as doing:

> When Renovate performs lockFileMaintenance it deletes the lock file and runs the relevant package manager. That package manager creates a new lock file, where all dependency versions are updated to the latest version. Renovate then commits that lock file to the update branch and creates the lock file update PR.

I wonder what makes it do this 🤔

**Comment by jerome-benoit** [comment]
> I wonder what makes it do this 🤔

Renovate does not run `npm dedupe` on PR maintenance.

**Comment by voxpelli** [comment]
> > I wonder what makes it do this 🤔
> 
> Renovate does not run `npm dedupe` on PR maintenance.

One can make it do so: https://docs.renovatebot.com/configuration-options/#postupdateoptions

But in this case the normal hoisting should be enough, the `npm dedupe` only differs by preferring sharing dependencies rather than newest dependencies, not sure why it caused duplication in this case

**Comment by jerome-benoit** [comment]
> > Renovate does not run `npm dedupe` on PR maintenance.
> 
> One can make it do so: https://docs.renovatebot.com/configuration-options/#postupdateoptions
> 

In that PR? 

> But in this case the normal hoisting should be enough, the `npm dedupe` only differs by preferring sharing dependencies rather than newest dependencies, not sure why it caused duplication in this case

Deduping algo is not triggered by default at install/update in npm/pnpm, probably for performance reason.

**Comment by voxpelli** [comment]
> Deduping algo is not triggered by default at install/update in npm/pnpm, probably for performance reason.

They themselves [note](https://docs.npmjs.com/cli/v10/commands/npm-dedupe):

> because npm favors updates by default, even when doing so causes duplication

I dug into why this duplication happens:

It's because [@typescript-eslint/utils](https://www.npmjs.com/package/@typescript-eslint/utils) has a `alpha40` release but [typescript-eslint](https://www.npmjs.com/package/typescript-eslint) is still at `alpha39` and `@stylistic/eslint-plugin-ts@2.6.0-beta.0` and `@stylistic/eslint-plugin-plus@2.6.0-beta.0` do like us and and pull in the very latest alpha release of `@typescript-eslint/utils` and but `typescript-eslint@alpha39`  specifically requests `@typescript-eslint/utils@alpha39` and as such the two get two different versions

I think this is correct behavior, and what users of `neostandard` will also get.

I have asked in the `typescript-eslint` discord channel for version 8 how it comes that `typescript-eslint` is out of sync

**Comment by jerome-benoit** [comment]
> 
> I think this is correct behavior, and what users of `neostandard` will also get.
> 

The definition of `correct behavior` here differs between package manager. 

And the behavior that enforce: 

- lock file minium size 
- prefer latest leaf dependencies version when possible by default

will fit a space efficient package management design. `pnpm` and `yarn` behaves that way is most case, `npm` not.  

Example: user of `neostandard` with `pnpm` will get a deduped local lock file while `npm` user will get a needlessly oversized lock file with unneeded deps dups. Using `npm dedupe` makes npm behaves closer to `pnpm`.

**Comment by voxpelli** [comment]
But since the lockfile isn't published it only serves as a reference to the npm based tests that are being run in this repository, to ensure that we catch regressions, so it doesn't really matter here, right?

**Comment by jerome-benoit** [comment]
> But since the lockfile isn't published it only serves as a reference to the npm based tests that are being run in this repository, to ensure that we catch regressions, so it doesn't really matter here, right?

How many times `npm i` or `npm ci` is run in CI? How many run are cached? What is the installation time between run with and without deduping? What is the cache total size with and without deduping? 

Since the CI in that repo is doing deps installation intensively, I would argue that using a space efficient package manager will make it greener :)

As a side note: you can't imagine the amount of space on SCM hosting infra that will be saved if npm lock file were deduped. 

I will revert the package-lock.json changes to avoid blocking the other change. And open a ticket/discussion on a space efficient CI when time permits.

**Review by voxpelli — APPROVED** [review]

**Pull request merged**